src/components/Header.astro

```diff
@@ -0,0 +1,41 @@
+---
+import MobileHeader from "./Mobile-header.astro";
+import DesktopHeader from "./Desktop-header.astro";
+---
+<script>
+    let lastScrollTop = 0;
+    const header = document.querySelectorAll(".header")
+    window.addEventListener("scroll", function() {
+        if(header === null) {
+            header = document.querySelectorAll(".header")
+        }
+        const currentScroll = window.scrollY || document.documentElement.scrollTop;
+        if (currentScroll > lastScrollTop) {
+            // Scroll down
+            header.forEach(header => header.classList.add("header-hidden"));
+        } else {
+            // Scroll up
+            header.forEach(header => header.classList.remove("header-hidden"));
+        }
+        lastScrollTop = currentScroll <= 0 ? 0 : currentScroll; // For Mobile or negative scrolling
+    }, false);
+
+    document.addEventListener("DOMContentLoaded", function() {
+        const currentPage = window.location.pathname; // Get current page pathname
+        // Get all subpage links
+        const links = document.querySelectorAll(".header nav ul li a");
```

Comment:
we can get reference to this element once before we add event listener same as in scroll